packages/php/email-editor/src/Integrations/Core/Renderer/Blocks/class-gallery.php

```
@@ -0,0 +1,267 @@
+		// Build the email-friendly layout.
+		return $this->build_email_layout( $gallery_images, $parsed_block, $block_content, $rendering_context );
+	}
+
+	/**
+	 * Extract all images from gallery content with their links and captions.
+	 *
+	 * @param string $block_content The rendered gallery block HTML.
+	 * @param array  $parsed_block The parsed block data.
+	 * @return array Array of sanitized image HTML strings.
+	 */
+	private function extract_images_from_gallery_content( string $block_content, array $parsed_block ): array {
```

PHPMD (2.15.0)

Avoid unused parameters such as '$block_content'. (undefined)

(UnusedFormalParameter)